nbgrader/apps/releaseapp.py

```diff
@@ -0,0 +1,97 @@
+import os
+import sys
+import shutil
+
+from IPython.utils.traitlets import Unicode, List, Bool
+
+from nbgrader.apps.baseapp import TransferApp, transfer_aliases, transfer_flags
+from nbgrader.utils import self_owned
+
+
+aliases = {}
+aliases.update(transfer_aliases)
+aliases.update({
+})
+
+flags = {}
+flags.update(transfer_flags)
+flags.update({
+    'force': (
+        {'ReleaseApp' : {'force' : True}},
+        "Force overwrite of existing files in the exchange."
+    ),
+    'remove': (
+        {'ReleaseApp' : {'remove': True}},
+        "Only remove existing files in the exchange."
+    ),
+})
+
+class ReleaseApp(TransferApp):
+
+    name = u'nbgrader-release'
+    description = u'Release an assignment to the nbgrader exchange'
+
+    aliases = aliases
+    flags = flags
+
+    examples = """
+        Here we go...
+        """
+
+    force = Bool(False, config=True, help="Force overwrite existing files in the exchange.")
+
+    remove = Bool(False, config=True, help="Only remove existing files in the exchange.")
```

Comment:
Ah, I like this. I would maybe clarify in the help string though that this is akin to "un-releasing" an assignment.